control-plane/pkg/reconciler/channel/channel.go

```diff
@@ -562,6 +576,21 @@ func (r *Reconciler) getChannelContractResource(ctx context.Context, topic strin
 	return resource, nil
 }
 
+func (r *Reconciler) getSubscriptionName(channel *messagingv1beta1.KafkaChannel, subscriber *v1.SubscriberSpec) (string, error) {
+	subscriptions, err := r.SubscriptionLister.Subscriptions(channel.GetNamespace()).List(labels.Everything())
+	if err != nil {
+		return "", fmt.Errorf("failed to list subscriptions in namespace %s: %w", channel.GetNamespace(), err)
+	}
+
+	for _, s := range subscriptions {
+		if s.UID == subscriber.UID {
+			return s.Name, nil
+		}
+	}
+
+	return "", errors.New(string(metav1.StatusReasonNotFound))
+}
+
```

Comment:
@aliok do you have a better/faster way of doing this? Do we have a label to list subscriptions for the given channel only?

Reply:
I checked and we don't have a label for this.